src/alpaka/AlpakaCore/alpakaWorkDivHelper.h

```diff
@@ -30,12 +30,12 @@ namespace cms {
 
     /*
      * Computes the range of the element(s) global index(es) in grid.
+     * Warning: the max index is not truncated by any max number of elements.
      */
-    template <typename T_Acc, typename T_Dim>
-    ALPAKA_FN_ACC std::pair<Vec<T_Dim>, Vec<T_Dim>> element_global_index_range(const T_Acc& acc,
-                                                                               const Vec<T_Dim>& maxNumberOfElements) {
+    template <typename T_Acc, typename T_Dim = alpaka::dim::Dim<T_Acc>>
+    ALPAKA_FN_ACC std::pair<Vec<T_Dim>, Vec<T_Dim>> element_global_index_range_uncut(const T_Acc& acc) {
```

Comment:
I would name this function as `element_global_index_range`, and the other something along `element_global_index_range_max` (preferring to tell what a function does rather than what it does not do). But a minor point for a prototype.

Reply:
ok no pb
yes was to insist on the risky version (the one which has not truncation with max), but yes the other way round can be more elegant. Will change it.